modules/core/src/main/java/org/apache/ignite/internal/commandline/cache/CheckIndexInlineSizes.java

```diff
@@ -0,0 +1,176 @@
+                .filter(SRV_NODES)
+                .collect(toSet());
+
+            Set<GridClientNode> supportedServerNodes = serverNodes.stream()
+                .filter(n -> checkIndexInlineSizesSupported(n))
+                .collect(toSet());
+
+            Collection<UUID> serverNodeIds = F.transform(serverNodes, GridClientNode::nodeId);
+            Collection<UUID> supportedServerNodeIds = F.transform(supportedServerNodes, GridClientNode::nodeId);
+
+            CheckIndexInlineSizesResult res = client.compute().projection(supportedServerNodes).execute(
+                CheckIndexInlineSizesTask.class.getName(),
+                new VisorTaskArgument<>(supportedServerNodeIds, false)
+            );
+
+            Set<UUID> unsupportedNodes = serverNodeIds.stream()
+                .filter(n -> !supportedServerNodeIds.contains(n))
+                .collect(toSet());
+
+            analizeResults(log, unsupportedNodes, res);
+        }
+
+        return null;
+    }
+
+    /** */
+    private void analizeResults(
+        Logger log,
+        Set<UUID> unsupportedNodes,
+        CheckIndexInlineSizesResult res
+    ) {
+        if (!F.isEmpty(unsupportedNodes))
+            log.info(String.format(NOT_ALL_NODES_SUPPORT_FEATURE_WARN_MSG_FMT, unsupportedNodes));
+
+        Map<String, Map<Integer, Set<UUID>>> indexToSizeNode = new HashMap<>();
+
+        for (Map.Entry<UUID, Map<String, Integer>> nodeRes : res.inlineSizes().entrySet()) {
+            for (Map.Entry<String, Integer> index : nodeRes.getValue().entrySet()) {
+                Map<Integer, Set<UUID>> sizeToNodes = indexToSizeNode.computeIfAbsent(index.getKey(), x -> new HashMap<>());
+
+                sizeToNodes.computeIfAbsent(index.getValue(), x -> new HashSet<>()).add(nodeRes.getKey());
+            }
+        }
+
+        log.info("Found " + indexToSizeNode.size() + " secondary indexes.");
+
+        Map<String, Map<Integer, Set<UUID>>> problems = indexToSizeNode.entrySet().stream()
+            .filter(e -> e.getValue().size() > 1)
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        if (F.isEmpty(problems))
+            log.info(INDEXES_INLINE_SIZE_ARE_SAME);
+        else
+            printProblemsAndShowRecommendations(problems, log);
+    }
+
+    /** */
+    private void printProblemsAndShowRecommendations(Map<String, Map<Integer, Set<UUID>>> problems, Logger log) {
+        log.info(problems.size() +
+            " indexes have different effective inline size on nodes. It can lead to performance degradation in SQL queries.");
+        log.info("Indexes:");
+
+        for (Map.Entry<String, Map<Integer, Set<UUID>>> entry : problems.entrySet()) {
+            SB sb = new SB();
+
+            sb.a("Full index name: ").a(entry.getKey());
+
+            for (Integer size : entry.getValue().keySet())
+                sb.a(" nodes: ").a(entry.getValue().get(size)).a(" inline size: ").a(size).a(",");
```

Comment:
column view here looks like more readable than string representation.

Reply:
can someone understand what nodes are talking about ?

Reply:
> column view here looks like more readable than string representation.
I'm not sure. I made a usability shield in the ticket's comments. CS, QA and Core teams are ok with current output. I'd prefer to keep it as is.

Reply:
> can someone understand what nodes are talking about ?

It can be helpful for investigation problems. Also, the user can check IGNITE_MAX_INDEX_PAYLOAD_SIZE property on the selected nodes.